devito/symbolics/extended_sympy.py

```diff
@@ -701,6 +696,31 @@ def rfunc(func, item, *args):
 }
 
 rfunc_mapper = {
-    min: MIN,
-    max: MAX,
+    min: Min,
+    max: Max,
 }
+
+
+def integer_args(*args):
+    """
+    Check if expression is Integer.
+    Used to choose the function printed in the c-code
+    """
+    if len(args) == 0:
+        return False
+
+    if len(args) == 1:
+        try:
+            return np.issubdtype(args[0].dtype, np.integer)
+        except AttributeError:
+            return args[0].is_integer
+    res = True
+    for a in args:
+        try:
+            if len(a.args) > 0:
+                res = res and integer_args(*a.args)
+            else:
+                res = res and integer_args(a)
+        except AttributeError:
+            res = res and integer_args(a)
+    return res
```

Comment:
So the implementation of this recursive function is a bit odd

What I would do (haven't tested the code though)

```
def is_integer_expr(expr):
    if expr.is_integer:
        return True

    try:
        return np.issubdtype(expr.dtype, np.integer)
    except AttributeError:
        pass

     # We must look inside the args
     return all(is_integer_expr(a) for a in expr.args)
```

Reply:
Because we have nested function with a lot of case that have `is_integer` defined but returns `None` because it's a function/expression so it returns wrong result with the simple recursive (had it exactly like that at initially)